annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/AnnotationController.java

```diff
@@ -314,29 +335,14 @@ private IsoformOverride getIsoformOverride(String source, String transcriptId)
 
     private List<Hotspot> getHotspotAnnotation(TranscriptConsequence transcript)
     {
-        List<Hotspot> hotspots = new ArrayList<>();
-        String transcriptId = transcript.getTranscriptId();
+        //String transcriptId = transcript.getTranscriptId();
         //Hotspot hotspot = hotspotRepository.findOne(transcriptId);
 
         // get the hotspot(s) from the web service
-        for (Hotspot hotspot : hotspotService.getHotspots(transcriptId))
-        {
-            if (Numerical.overlaps(hotspot.getResidue(),
-                       transcript.getProteinStart(),
-                       transcript.getProteinEnd()))
-            {
-                // TODO use a JSON view instead of copying fields to another model?
-                // we have data duplication here...
-                hotspot.setGeneId(transcript.getGeneId());
-                hotspot.setProteinStart(transcript.getProteinStart());
-                hotspot.setProteinEnd(transcript.getProteinEnd());
+        List<Hotspot> hotspots = hotspotService.getHotspots(transcript);
 
-                hotspots.add(hotspot);
-
-                // do not cache anything for now
-                //hotspotRepository.save(hotspots);
-            }
-        }
+        // do not cache anything for now
+        //hotspotRepository.save(hotspots);
```

Comment:
Initially we were planning to query the cancerhotspots API for each transcript separately, and then cache the result in the genome-nexus database, but later we decided to retrieve all hotspot mutations at once and cache them in memory. So, currently the hotspot repository is not used.